pkg/api/provision.go

```diff
@@ -324,9 +324,19 @@ func (s *server) isParentProvisioning(instance service.Instance) (bool, error) {
 		return false, err
 	}
 
-	//Parent has was not found, so wait for that that to occur
+	// Parent is not found, we treat this as a bad provision request
 	if !parentFound {
-		return true, nil
+		return false, fmt.Errorf(
```

Comment:
This was done to support the way things can be done with Kubernetes. Take a Helm chart that may declare both a parent and child instance. When these are created, Kubernetes via Service Catalog will make the request for both resources, but there isn't a way to constrain the order. Therefore, you could get the child before the parent. In this scenario, the change you've introduced would cause things to end up in a failure state.